File: units/RocketLauncher.go
package units

type RocketLauncher struct {
	CombatUnit
}

func NewRocketLauncher() *RocketLauncher {
	rocketLauncher := new(RocketLauncher)
	rocketLauncher.OgameID = 401
	rocketLauncher.Name = "RocketLauncher"
	rocketLauncher.Price = Price{2000, 0, 0}
	rocketLauncher.ShieldPower = 20
	rocketLauncher.WeaponPower = 80
	rocketLauncher.InitHullPlating()
	rocketLauncher.RestoreShield()
	rocketLauncher.RapidfireFrom = map[string]int{"Bomber": 20, "Cruiser": 10, "Deathstar": 200}
	return rocketLauncher
}

File: units/RocketLauncher.py
from units.Units import CombatUnit
from units.Units import Price

class RocketLauncher(CombatUnit):
    OgameID = 401
    Name = "RocketLauncher"
    Price = Price(2000, 0, 0)
    ShieldPower = 20
    WeaponPower = 80
    RapidfireFrom = {"Bomber": 20, "Cruiser": 10, "Deathstar": 200}

    def __init__(self):
        self.InitHullPlating()
        self.RestoreShield()
